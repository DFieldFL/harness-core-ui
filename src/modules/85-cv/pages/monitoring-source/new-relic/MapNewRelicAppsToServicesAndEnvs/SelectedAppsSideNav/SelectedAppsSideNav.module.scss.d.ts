/* eslint-disable */
// this is an auto-generated file
declare const styles: {
  readonly main: string
  readonly navHeader: string
  readonly seletedAppContainer: string
  readonly selectedApp: string
}
export default styles
