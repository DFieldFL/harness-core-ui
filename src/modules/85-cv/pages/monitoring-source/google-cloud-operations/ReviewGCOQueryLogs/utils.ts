import type { GCOMonitoringSourceInfo } from '../GoogleCloudOperationsMonitoringSourceUtils'
import type { GCOLogsDSConfig, MapGCOLogsQueryToService } from '../MapQueriesToHarnessService/types'

export function transformGCOLogsToDSConfig(setupSource: GCOMonitoringSourceInfo): GCOLogsDSConfig {
  const dsConfig: GCOLogsDSConfig = {
    connectorIdentifier: setupSource.connectorRef?.value as string,
    type: 'STACKDRIVER_LOG',
    accountId: setupSource.accountId,
    projectIdentifier: setupSource.projectIdentifier,
    orgIdentifier: setupSource.orgIdentifier,
    identifier: setupSource.identifier,
    monitoringSourceName: setupSource.name,
    logConfigurations: []
  }

  for (const entry of setupSource.mappedServicesAndEnvs.entries()) {
    const {
      metricName,
      query,
      serviceIdentifier,
      envIdentifier,
      serviceInstance,
      messageIdentifier
    }: MapGCOLogsQueryToService = entry[1]

    dsConfig.logConfigurations.push({
      envIdentifier: envIdentifier?.value as string,
      serviceIdentifier: serviceIdentifier?.value as string,
      logDefinition: {
        name: metricName,
        query,
        serviceInstanceIdentifier: serviceInstance,
        messageIdentifier: messageIdentifier
      }
    })
  }

  return dsConfig
}
