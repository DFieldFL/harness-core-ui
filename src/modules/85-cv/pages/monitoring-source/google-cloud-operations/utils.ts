import type { ProjectPathProps } from '@common/interfaces/RouteInterfaces'
import {
  GCODSConfig,
  GCOMonitoringSourceInfo,
  buildGCOMonitoringSourceInfo,
  GCOProduct
} from './GoogleCloudOperationsMonitoringSourceUtils'
import type { GCOLogsDSConfig } from './MapQueriesToHarnessService/types'

export function transformGetResponseForStackDriver(
  gcoConfig: GCODSConfig,
  params: ProjectPathProps & { identifier: string }
): GCOMonitoringSourceInfo {
  const gcoInfo: GCOMonitoringSourceInfo = buildGCOMonitoringSourceInfo(params, GCOProduct.CLOUD_METRICS)
  if (!gcoConfig) {
    return gcoConfig
  }

  gcoInfo.accountId = gcoConfig.accountId
  gcoInfo.identifier = gcoConfig.identifier
  gcoInfo.name = gcoConfig.monitoringSourceName
  gcoInfo.type = 'STACKDRIVER'
  gcoInfo.orgIdentifier = gcoConfig.orgIdentifier
  gcoInfo.projectIdentifier = gcoConfig.projectIdentifier
  gcoInfo.isEdit = !!gcoConfig.identifier
  gcoInfo.connectorRef = {
    label: gcoConfig.connectorIdentifier as string,
    value: gcoConfig.connectorIdentifier as string
  }
  gcoInfo.product = GCOProduct.CLOUD_METRICS
  gcoInfo.selectedDashboards = []
  gcoInfo.selectedMetrics = new Map()

  for (const config of gcoConfig.metricConfigurations) {
    const { envIdentifier, serviceIdentifier, metricDefinition } = config
    // TODO: Remove this once backend fixes discripancy in key names manualQuery and isManualQuery
    const manualQuery = metricDefinition?.isManualQuery || (metricDefinition as any)?.manualQuery
    if (
      !envIdentifier ||
      !serviceIdentifier ||
      !metricDefinition ||
      !metricDefinition.dashboardName ||
      !(manualQuery ? true : metricDefinition.dashboardPath) ||
      !metricDefinition.metricName
    )
      continue

    const metricTags: any = {}
    for (const tag of metricDefinition.metricTags || []) {
      metricTags[tag] = ''
    }

    gcoInfo.selectedDashboards.push({ name: metricDefinition.dashboardName, path: metricDefinition.dashboardPath })
    gcoInfo.selectedMetrics.set(metricDefinition.metricName, {
      metricName: metricDefinition.metricName,
      metricTags,
      dashboardName: metricDefinition.dashboardName,
      dashboardPath: metricDefinition.dashboardPath,
      service: { label: serviceIdentifier, value: serviceIdentifier },
      environment: { label: envIdentifier, value: envIdentifier },
      isManualQuery: manualQuery,
      riskCategory: `${metricDefinition.riskProfile?.category}/${metricDefinition.riskProfile?.metricType}`,
      higherBaselineDeviation: metricDefinition.riskProfile?.thresholdTypes?.includes('ACT_WHEN_HIGHER'),
      lowerBaselineDeviation: metricDefinition.riskProfile?.thresholdTypes?.includes('ACT_WHEN_LOWER'),
      query: JSON.stringify(metricDefinition.jsonMetricDefinition, null, 2)
    })
  }

  return gcoInfo
}

export function transformGetResponseForStackDriverLogs(
  gcoLogsConfig: GCOLogsDSConfig,
  params: ProjectPathProps & { identifier: string }
): GCOMonitoringSourceInfo {
  const gcoLogsInfo: GCOMonitoringSourceInfo = buildGCOMonitoringSourceInfo(params, GCOProduct.CLOUD_LOGS)
  if (!gcoLogsConfig) {
    return gcoLogsConfig
  }

  gcoLogsInfo.accountId = gcoLogsConfig.accountId
  gcoLogsInfo.identifier = gcoLogsConfig.identifier
  gcoLogsInfo.name = gcoLogsConfig.monitoringSourceName
  gcoLogsInfo.type = 'STACKDRIVER_LOG'
  gcoLogsInfo.orgIdentifier = gcoLogsConfig.orgIdentifier
  gcoLogsInfo.projectIdentifier = gcoLogsConfig.projectIdentifier
  gcoLogsInfo.connectorRef = {
    label: gcoLogsConfig.connectorIdentifier as string,
    value: gcoLogsConfig.connectorIdentifier as string
  }
  gcoLogsInfo.product = GCOProduct.CLOUD_LOGS
  gcoLogsInfo.isEdit = !!gcoLogsInfo.identifier

  for (const config of gcoLogsConfig.logConfigurations) {
    const { envIdentifier, serviceIdentifier, logDefinition } = config
    if (
      !envIdentifier ||
      !serviceIdentifier ||
      !logDefinition ||
      !logDefinition.name ||
      !logDefinition.query ||
      !logDefinition.serviceInstanceIdentifier ||
      !logDefinition.messageIdentifier
    ) {
      continue
    }

    gcoLogsInfo.mappedServicesAndEnvs.set(logDefinition.name, {
      metricName: logDefinition.name,
      serviceIdentifier: { label: serviceIdentifier, value: serviceIdentifier },
      envIdentifier: { label: envIdentifier, value: envIdentifier },
      serviceInstance: logDefinition.serviceInstanceIdentifier,
      messageIdentifier: logDefinition.messageIdentifier,
      query: logDefinition.query
    })
  }

  return gcoLogsInfo
}
