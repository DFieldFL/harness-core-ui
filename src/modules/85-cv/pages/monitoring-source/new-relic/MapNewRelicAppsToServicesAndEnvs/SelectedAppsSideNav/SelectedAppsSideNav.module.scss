.main {
  padding: var(--spacing-medium) !important;
  padding-left: var(--spacing-xlarge) !important;
  overflow-y: scroll;

  .navHeader {
    margin-bottom: var(--spacing-medium) !important;
    color: var(--black) !important;
  }

  .seletedAppContainer {
    padding: var(--spacing-small) 0 !important;
    border-top: 1px solid var(--grey-300);
    .selectedApp {
      font-size: var(--font-size-medium) !important;
      color: var(--black) !important;
      border-left: 3px solid var(--grey-300);
      padding-left: var(--spacing-small) !important;
    }
  }
}
