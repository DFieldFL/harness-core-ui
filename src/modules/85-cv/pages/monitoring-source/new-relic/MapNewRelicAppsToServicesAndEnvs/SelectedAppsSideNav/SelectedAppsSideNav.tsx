import React from 'react'
import { Container, Text } from '@wings-software/uicore'
import { Classes } from '@blueprintjs/core'
import cx from 'classnames'
import { useStrings } from 'framework/strings'
import css from './SelectedAppsSideNav.module.scss'

const LoadingCells = [1, 2, 3, 4, 5]

export interface SelectedAppsSideNavProps {
  selectedApps?: string[]
  loading?: boolean
}

export function SelectedAppsSideNav(props: SelectedAppsSideNavProps): JSX.Element {
  const { selectedApps, loading } = props
  const { getString } = useStrings()
  return (
    <Container className={css.main}>
      <Text className={css.navHeader}>{getString('cv.monitoringSources.newRelic.selectedApplications')}</Text>
      {loading
        ? LoadingCells.map(loadingCell => (
            <Container key={loadingCell} className={css.seletedAppContainer}>
              <Container className={cx(Classes.SKELETON, css.selectedApp)} height={16} width="100%" />
            </Container>
          ))
        : selectedApps?.map(selectedApp => {
            return (
              <Container key={selectedApp} className={css.seletedAppContainer}>
                <Text className={css.selectedApp}>{selectedApp}</Text>
              </Container>
            )
          })}
    </Container>
  )
}
