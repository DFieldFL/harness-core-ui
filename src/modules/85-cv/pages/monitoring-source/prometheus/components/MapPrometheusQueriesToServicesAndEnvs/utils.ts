import type { SelectOption } from '@wings-software/uicore'
import { isNumber } from 'lodash-es'
import type { FormikProps } from 'formik'
import type { UseStringsReturn } from 'framework/strings'
import type { MapPrometheusQueryToService } from '../../constants'
import { MapPrometheusQueryToServiceFieldNames } from './constants'

type UpdateSelectedMetricsMap = {
  updatedMetric: string
  oldMetric: string
  mappedMetrics: Map<string, MapPrometheusQueryToService>
  formikProps: FormikProps<MapPrometheusQueryToService | undefined>
}

export function updateSelectedMetricsMap({
  updatedMetric,
  oldMetric,
  mappedMetrics,
  formikProps
}: UpdateSelectedMetricsMap) {
  const updatedMap = new Map(mappedMetrics)

  // in the case where user updates metric name, update the key for current value
  if (oldMetric !== formikProps.values?.metricName) {
    updatedMap.delete(oldMetric)
  }

  // if newly created metric create form object
  if (!updatedMap.has(updatedMetric)) {
    updatedMap.set(updatedMetric, { metricName: updatedMetric, query: '', isManualQuery: false })
  }

  // update map with current form data
  if (formikProps.values?.metricName) {
    updatedMap.set(formikProps.values.metricName, formikProps.values as MapPrometheusQueryToService)
  }
  return { selectedMetric: updatedMetric, mappedMetrics: updatedMap }
}

export function validateMappings(
  getString: UseStringsReturn['getString'],
  createdMetrics: string[],
  selectedMetricIndex: number,
  values?: MapPrometheusQueryToService
): { [fieldName: string]: string } {
  const requiredFieldErrors = {
    [MapPrometheusQueryToServiceFieldNames.ENVIRONMENT]: getString('cv.monitoringSources.envValidation'),
    [MapPrometheusQueryToServiceFieldNames.ENVIRONMENT_FILTER]: getString(
      'cv.monitoringSources.prometheus.validation.filterOnEnvironment'
    ),
    [MapPrometheusQueryToServiceFieldNames.METRIC_NAME]: getString('cv.monitoringSources.metricNameValidation'),
    [MapPrometheusQueryToServiceFieldNames.PROMETHEUS_METRIC]: getString(
      'cv.monitoringSources.prometheus.validation.promethusMetric'
    ),
    [MapPrometheusQueryToServiceFieldNames.RISK_CATEGORY]: getString(
      'cv.monitoringSources.gco.mapMetricsToServicesPage.validation.riskCategory'
    ),
    [MapPrometheusQueryToServiceFieldNames.SERVICE]: getString('cv.monitoringSources.serviceValidation'),
    [MapPrometheusQueryToServiceFieldNames.SERVICE_FILTER]: getString(
      'cv.monitoringSources.prometheus.validation.filterOnService'
    ),
    [MapPrometheusQueryToServiceFieldNames.SERVICE_INSTANCE]: getString(
      'cv.monitoringSources.prometheus.validation.serviceInstanceIdentifier'
    ),
    [MapPrometheusQueryToServiceFieldNames.QUERY]: getString(
      'cv.monitoringSources.gco.manualInputQueryModal.validation.query'
    )
  }

  if (!values) {
    return requiredFieldErrors
  }

  if (values.isManualQuery) {
    delete requiredFieldErrors[MapPrometheusQueryToServiceFieldNames.SERVICE_FILTER]
    delete requiredFieldErrors[MapPrometheusQueryToServiceFieldNames.ENVIRONMENT_FILTER]
    delete requiredFieldErrors[MapPrometheusQueryToServiceFieldNames.SERVICE_INSTANCE]
    delete requiredFieldErrors[MapPrometheusQueryToServiceFieldNames.PROMETHEUS_METRIC]
  }

  for (const fieldName of Object.keys(requiredFieldErrors)) {
    switch (fieldName) {
      case MapPrometheusQueryToServiceFieldNames.ENVIRONMENT_FILTER:
        if (values.envFilter?.length) delete requiredFieldErrors[fieldName]
        break
      case MapPrometheusQueryToServiceFieldNames.SERVICE_FILTER:
        if (values.serviceFilter?.length) delete requiredFieldErrors[fieldName]
        break
      default:
        if ((values as any)[fieldName]) delete requiredFieldErrors[fieldName]
    }
  }

  if (values.lowerBaselineDeviation !== true && values.higherBaselineDeviation !== true) {
    requiredFieldErrors[MapPrometheusQueryToServiceFieldNames.LOWER_BASELINE_DEVIATION] = getString(
      'cv.monitoringSources.prometheus.validation.deviation'
    )
  }

  const duplicateNames = createdMetrics?.filter((metricName, index) => {
    if (index === selectedMetricIndex) {
      return false
    }
    return metricName === values.metricName
  })

  if (values.metricName && duplicateNames.length) {
    requiredFieldErrors[MapPrometheusQueryToServiceFieldNames.METRIC_NAME] = getString(
      'cv.monitoringSources.prometheus.validation.metricNameUnique'
    )
  }

  if (
    !requiredFieldErrors[MapPrometheusQueryToServiceFieldNames.QUERY] &&
    isNumber(values.recordCount) &&
    values.recordCount > 1
  ) {
    requiredFieldErrors[MapPrometheusQueryToServiceFieldNames.QUERY] = getString(
      'cv.monitoringSources.prometheus.validation.recordCount'
    )
  }

  return requiredFieldErrors
}

export function initializePrometheusGroupNames(
  mappedMetrics: Map<string, MapPrometheusQueryToService>,
  getString: UseStringsReturn['getString']
): SelectOption[] {
  const groupNames = Array.from(mappedMetrics?.entries())
    .map(metric => {
      const { groupName } = metric?.[1] || {}
      return groupName || null
    })
    .filter(groupItem => groupItem !== null) as SelectOption[]
  return [{ label: getString('cv.addNew'), value: '' }, ...groupNames]
}
