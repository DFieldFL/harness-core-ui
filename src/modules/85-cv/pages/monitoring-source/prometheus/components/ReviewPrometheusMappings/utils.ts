import type { MultiSelectOption } from '@wings-software/uicore'
import type { MetricDefinition, TimeSeriesMetricDefinition } from 'services/cv'

import type {
  PrometheusDSConfig,
  PrometheusSetupSource,
  MapPrometheusQueryToService,
  RiskProfileCatgory,
  PrometheusFilter
} from '../../constants'

export function transformLabelToPrometheusFilter(options?: MultiSelectOption[]): PrometheusFilter[] {
  const filters: PrometheusFilter[] = []
  if (!options?.length) {
    return filters
  }

  for (const option of options) {
    const value = option.label.split(':')
    filters.push({ labelName: value[0], labelValue: value[1] })
  }

  return filters
}

export function transformPrometheusSourceToDSConfig(setupSource: PrometheusSetupSource): PrometheusDSConfig {
  const dsConfig: PrometheusDSConfig = {
    connectorIdentifier: setupSource.connectorRef?.value,
    type: 'PROMETHEUS',
    accountId: setupSource.accountId,
    projectIdentifier: setupSource.projectIdentifier,
    orgIdentifier: setupSource.orgIdentifier,
    productName: setupSource.productName,
    identifier: setupSource.identifier,
    monitoringSourceName: setupSource.monitoringSourceName,
    metricDefinitions: []
  }

  for (const entry of setupSource.mappedServicesAndEnvs.entries()) {
    const {
      envIdentifier,
      envFilter,
      metricName,
      groupName,
      prometheusMetric,
      serviceFilter,
      serviceInstance,
      serviceIdentifier,
      isManualQuery,
      query,
      aggregator,
      additionalFilter,
      riskCategory,
      lowerBaselineDeviation,
      higherBaselineDeviation
    }: MapPrometheusQueryToService = entry[1]

    if (!envIdentifier || !serviceIdentifier || !groupName || !metricName || !riskCategory) {
      continue
    }

    if (!isManualQuery && (!prometheusMetric || !envFilter || !serviceFilter || !serviceInstance)) {
      continue
    }

    const [category, metricType] = riskCategory?.split('/') || []
    const thresholdTypes: TimeSeriesMetricDefinition['thresholdType'][] = []
    if (lowerBaselineDeviation) {
      thresholdTypes.push('ACT_WHEN_LOWER')
    }
    if (higherBaselineDeviation) {
      thresholdTypes.push('ACT_WHEN_HIGHER')
    }

    dsConfig.metricDefinitions.push({
      prometheusMetric,
      metricName,
      serviceFilter: transformLabelToPrometheusFilter(serviceFilter),
      isManualQuery,
      query,
      envFilter: transformLabelToPrometheusFilter(envFilter),
      additionalFilters: transformLabelToPrometheusFilter(additionalFilter),
      aggregation: aggregator,
      serviceInstanceFieldName: serviceInstance,
      groupName: groupName.value as string,
      envIdentifier: envIdentifier.value as string,
      serviceIdentifier: serviceIdentifier.value as string,
      riskProfile: {
        category: category as RiskProfileCatgory,
        metricType: metricType as MetricDefinition['type'],
        thresholdTypes
      }
    })
  }

  return dsConfig
}
