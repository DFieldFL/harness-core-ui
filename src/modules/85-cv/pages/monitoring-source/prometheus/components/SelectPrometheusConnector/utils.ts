import * as Yup from 'yup'
import type { DefineYourMonitoringSourceProps } from '@cv/components/CVSetupSourcesView/DefineYourMonitoringSource/DefineYourMonitoringSource'
import type { UseStringsReturn } from 'framework/strings'
import { StringUtils } from '@common/exports'

export function getDefineMonitoringSourceBaseProps(
  getString: UseStringsReturn['getString']
): Omit<DefineYourMonitoringSourceProps, 'mainHeading' | 'subHeading' | 'formikProps'> {
  return {
    stepLabelProps: {
      stepNumber: 1,
      totalSteps: 3
    },
    iconLabel: getString('connectors.prometheusLabel'),
    sourceIcon: { name: 'service-prometheus' }
  }
}

export function getValidationSchema(getString: UseStringsReturn['getString']): Record<string, any> {
  return Yup.object({
    monitoringSourceName: Yup.string().required(getString('cv.onboarding.selectProductScreen.validationText.name')),
    identifier: Yup.string().when('name', {
      is: val => val?.length,
      then: Yup.string()
        .trim()
        .required(getString('cv.onboarding.selectProductScreen.validationText.identifier'))
        .matches(
          /^(?![0-9])[0-9a-zA-Z_$]*$/,
          getString('cv.onboarding.selectProductScreen.validationText.validIdRegex')
        )
        .notOneOf(StringUtils.illegalIdentifiers)
    }),
    connectorRef: Yup.object().required(getString('cv.onboarding.selectProductScreen.validationText.connectorRef')),
    productName: Yup.string().required(getString('cv.onboarding.selectProductScreen.validationText.product'))
  })
}
