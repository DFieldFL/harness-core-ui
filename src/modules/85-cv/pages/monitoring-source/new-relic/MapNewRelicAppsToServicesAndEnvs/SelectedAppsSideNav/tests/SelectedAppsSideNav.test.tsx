import React from 'react'
import { Classes } from '@blueprintjs/core'
import { render, waitFor } from '@testing-library/react'
import { TestWrapper } from '@common/utils/testUtils'
import { SelectedAppsSideNav, SelectedAppsSideNavProps } from '../SelectedAppsSideNav'

function WrapperComponent(props: SelectedAppsSideNavProps): JSX.Element {
  return (
    <TestWrapper>
      <SelectedAppsSideNav {...props} />
    </TestWrapper>
  )
}

describe('Unit tests for SelectedAppsSideNav', () => {
  test('Ensure loading state is rendered correectly', async () => {
    const { container } = render(<WrapperComponent loading={true} />)
    await waitFor(() => expect(container.querySelectorAll(`[class*="${Classes.SKELETON}"]`).length).toBe(5))
  })
  test('Ensure that when apps are provided, the list is rendered', async () => {
    const { container } = render(
      <WrapperComponent
        selectedApps={Array(50)
          .fill(null)
          .map((_, i) => i.toString())}
      />
    )
    await waitFor(() => expect(container.querySelectorAll(`[class*="selectedApp"]`).length).toBe(50))
    expect(container).toMatchSnapshot()
  })
})
