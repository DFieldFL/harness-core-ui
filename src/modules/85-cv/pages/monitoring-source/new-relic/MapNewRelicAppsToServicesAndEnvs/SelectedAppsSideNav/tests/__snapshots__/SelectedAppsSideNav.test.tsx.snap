// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit tests for SelectedAppsSideNav Ensure that when apps are provided, the list is rendered 1`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main main"
  >
    <p
      class="StyledProps--font StyledProps--main navHeader"
    >
      cv.monitoringSources.newRelic.selectedApplications
    </p>
    <div
      class="StyledProps--font StyledProps--main seletedAppContainer"
    >
      <p
        class="StyledProps--font StyledProps--main selectedApp"
      >
        0
      </p>
    </div>
    <div
      class="StyledProps--font StyledProps--main seletedAppContainer"
    >
      <p
        class="StyledProps--font StyledProps--main selectedApp"
      >
        1
      </p>
    </div>
    <div
      class="StyledProps--font StyledProps--main seletedAppContainer"
    >
      <p
        class="StyledProps--font StyledProps--main selectedApp"
      >
        2
      </p>
    </div>
    <div
      class="StyledProps--font StyledProps--main seletedAppContainer"
    >
      <p
        class="StyledProps--font StyledProps--main selectedApp"
      >
        3
      </p>
    </div>
    <div
      class="StyledProps--font StyledProps--main seletedAppContainer"
    >
      <p
        class="StyledProps--font StyledProps--main selectedApp"
      >
        4
      </p>
    </div>
    <div
      class="StyledProps--font StyledProps--main seletedAppContainer"
    >
      <p
        class="StyledProps--font StyledProps--main selectedApp"
      >
        5
      </p>
    </div>
    <div
      class="StyledProps--font StyledProps--main seletedAppContainer"
    >
      <p
        class="StyledProps--font StyledProps--main selectedApp"
      >
        6
      </p>
    </div>
    <div
      class="StyledProps--font StyledProps--main seletedAppContainer"
    >
      <p
        class="StyledProps--font StyledProps--main selectedApp"
      >
        7
      </p>
    </div>
    <div
      class="StyledProps--font StyledProps--main seletedAppContainer"
    >
      <p
        class="StyledProps--font StyledProps--main selectedApp"
      >
        8
      </p>
    </div>
    <div
      class="StyledProps--font StyledProps--main seletedAppContainer"
    >
      <p
        class="StyledProps--font StyledProps--main selectedApp"
      >
        9
      </p>
    </div>
    <div
      class="StyledProps--font StyledProps--main seletedAppContainer"
    >
      <p
        class="StyledProps--font StyledProps--main selectedApp"
      >
        10
      </p>
    </div>
    <div
      class="StyledProps--font StyledProps--main seletedAppContainer"
    >
      <p
        class="StyledProps--font StyledProps--main selectedApp"
      >
        11
      </p>
    </div>
    <div
      class="StyledProps--font StyledProps--main seletedAppContainer"
    >
      <p
        class="StyledProps--font StyledProps--main selectedApp"
      >
        12
      </p>
    </div>
    <div
      class="StyledProps--font StyledProps--main seletedAppContainer"
    >
      <p
        class="StyledProps--font StyledProps--main selectedApp"
      >
        13
      </p>
    </div>
    <div
      class="StyledProps--font StyledProps--main seletedAppContainer"
    >
      <p
        class="StyledProps--font StyledProps--main selectedApp"
      >
        14
      </p>
    </div>
    <div
      class="StyledProps--font StyledProps--main seletedAppContainer"
    >
      <p
        class="StyledProps--font StyledProps--main selectedApp"
      >
        15
      </p>
    </div>
    <div
      class="StyledProps--font StyledProps--main seletedAppContainer"
    >
      <p
        class="StyledProps--font StyledProps--main selectedApp"
      >
        16
      </p>
    </div>
    <div
      class="StyledProps--font StyledProps--main seletedAppContainer"
    >
      <p
        class="StyledProps--font StyledProps--main selectedApp"
      >
        17
      </p>
    </div>
    <div
      class="StyledProps--font StyledProps--main seletedAppContainer"
    >
      <p
        class="StyledProps--font StyledProps--main selectedApp"
      >
        18
      </p>
    </div>
    <div
      class="StyledProps--font StyledProps--main seletedAppContainer"
    >
      <p
        class="StyledProps--font StyledProps--main selectedApp"
      >
        19
      </p>
    </div>
    <div
      class="StyledProps--font StyledProps--main seletedAppContainer"
    >
      <p
        class="StyledProps--font StyledProps--main selectedApp"
      >
        20
      </p>
    </div>
    <div
      class="StyledProps--font StyledProps--main seletedAppContainer"
    >
      <p
        class="StyledProps--font StyledProps--main selectedApp"
      >
        21
      </p>
    </div>
    <div
      class="StyledProps--font StyledProps--main seletedAppContainer"
    >
      <p
        class="StyledProps--font StyledProps--main selectedApp"
      >
        22
      </p>
    </div>
    <div
      class="StyledProps--font StyledProps--main seletedAppContainer"
    >
      <p
        class="StyledProps--font StyledProps--main selectedApp"
      >
        23
      </p>
    </div>
    <div
      class="StyledProps--font StyledProps--main seletedAppContainer"
    >
      <p
        class="StyledProps--font StyledProps--main selectedApp"
      >
        24
      </p>
    </div>
    <div
      class="StyledProps--font StyledProps--main seletedAppContainer"
    >
      <p
        class="StyledProps--font StyledProps--main selectedApp"
      >
        25
      </p>
    </div>
    <div
      class="StyledProps--font StyledProps--main seletedAppContainer"
    >
      <p
        class="StyledProps--font StyledProps--main selectedApp"
      >
        26
      </p>
    </div>
    <div
      class="StyledProps--font StyledProps--main seletedAppContainer"
    >
      <p
        class="StyledProps--font StyledProps--main selectedApp"
      >
        27
      </p>
    </div>
    <div
      class="StyledProps--font StyledProps--main seletedAppContainer"
    >
      <p
        class="StyledProps--font StyledProps--main selectedApp"
      >
        28
      </p>
    </div>
    <div
      class="StyledProps--font StyledProps--main seletedAppContainer"
    >
      <p
        class="StyledProps--font StyledProps--main selectedApp"
      >
        29
      </p>
    </div>
    <div
      class="StyledProps--font StyledProps--main seletedAppContainer"
    >
      <p
        class="StyledProps--font StyledProps--main selectedApp"
      >
        30
      </p>
    </div>
    <div
      class="StyledProps--font StyledProps--main seletedAppContainer"
    >
      <p
        class="StyledProps--font StyledProps--main selectedApp"
      >
        31
      </p>
    </div>
    <div
      class="StyledProps--font StyledProps--main seletedAppContainer"
    >
      <p
        class="StyledProps--font StyledProps--main selectedApp"
      >
        32
      </p>
    </div>
    <div
      class="StyledProps--font StyledProps--main seletedAppContainer"
    >
      <p
        class="StyledProps--font StyledProps--main selectedApp"
      >
        33
      </p>
    </div>
    <div
      class="StyledProps--font StyledProps--main seletedAppContainer"
    >
      <p
        class="StyledProps--font StyledProps--main selectedApp"
      >
        34
      </p>
    </div>
    <div
      class="StyledProps--font StyledProps--main seletedAppContainer"
    >
      <p
        class="StyledProps--font StyledProps--main selectedApp"
      >
        35
      </p>
    </div>
    <div
      class="StyledProps--font StyledProps--main seletedAppContainer"
    >
      <p
        class="StyledProps--font StyledProps--main selectedApp"
      >
        36
      </p>
    </div>
    <div
      class="StyledProps--font StyledProps--main seletedAppContainer"
    >
      <p
        class="StyledProps--font StyledProps--main selectedApp"
      >
        37
      </p>
    </div>
    <div
      class="StyledProps--font StyledProps--main seletedAppContainer"
    >
      <p
        class="StyledProps--font StyledProps--main selectedApp"
      >
        38
      </p>
    </div>
    <div
      class="StyledProps--font StyledProps--main seletedAppContainer"
    >
      <p
        class="StyledProps--font StyledProps--main selectedApp"
      >
        39
      </p>
    </div>
    <div
      class="StyledProps--font StyledProps--main seletedAppContainer"
    >
      <p
        class="StyledProps--font StyledProps--main selectedApp"
      >
        40
      </p>
    </div>
    <div
      class="StyledProps--font StyledProps--main seletedAppContainer"
    >
      <p
        class="StyledProps--font StyledProps--main selectedApp"
      >
        41
      </p>
    </div>
    <div
      class="StyledProps--font StyledProps--main seletedAppContainer"
    >
      <p
        class="StyledProps--font StyledProps--main selectedApp"
      >
        42
      </p>
    </div>
    <div
      class="StyledProps--font StyledProps--main seletedAppContainer"
    >
      <p
        class="StyledProps--font StyledProps--main selectedApp"
      >
        43
      </p>
    </div>
    <div
      class="StyledProps--font StyledProps--main seletedAppContainer"
    >
      <p
        class="StyledProps--font StyledProps--main selectedApp"
      >
        44
      </p>
    </div>
    <div
      class="StyledProps--font StyledProps--main seletedAppContainer"
    >
      <p
        class="StyledProps--font StyledProps--main selectedApp"
      >
        45
      </p>
    </div>
    <div
      class="StyledProps--font StyledProps--main seletedAppContainer"
    >
      <p
        class="StyledProps--font StyledProps--main selectedApp"
      >
        46
      </p>
    </div>
    <div
      class="StyledProps--font StyledProps--main seletedAppContainer"
    >
      <p
        class="StyledProps--font StyledProps--main selectedApp"
      >
        47
      </p>
    </div>
    <div
      class="StyledProps--font StyledProps--main seletedAppContainer"
    >
      <p
        class="StyledProps--font StyledProps--main selectedApp"
      >
        48
      </p>
    </div>
    <div
      class="StyledProps--font StyledProps--main seletedAppContainer"
    >
      <p
        class="StyledProps--font StyledProps--main selectedApp"
      >
        49
      </p>
    </div>
  </div>
</div>
`;
